signac/core/mongodbcollection.py

```diff
@@ -0,0 +1,147 @@
+
+        >>> doc['foo'] = dict(bar=True)
+        >>> doc
+        {'foo': {'bar': True}}
+        >>> doc.foo.bar = False
+        {'foo': {'bar': False}}
+
+    .. warning::
+
+        While the MongoDict object behaves like a dictionary, there are
+        important distinctions to remember. In particular, because operations
+        are reflected as changes to an underlying database, copying (even deep
+        copying) a MongoDict instance may exhibit unexpected behavior. If a
+        true copy is required, you should use the `to_base()` method to get a
+        dictionary representation, and if necessary construct a new MongoDict
+        instance: `new_dict = MongoDict(old_dict.to_base())`.
+
+    Parameters
+    ----------
+    name: str
+        The name of the  collection (Default value = None).
+    client: object
+        A Mongo client.
+    mongo_kwargs: dict
+        kwargs arguments passed through to the `pymongo.MongoClient` function.
+    database : string
+        Name of database (Default value = 'signac_db').
+    collection : string
+        Name of collection (Default value = 'collection')
+    data: mapping, optional
+        The intial data pass to MongoDict. Defaults to `dict()`
+    parent: object, optional
+        A parent instance of MongoDict or None (Default value = None).
+    """
+    pass
+
+
+class MongoList(MongoCollection, SyncedList):
+    """A non-string sequence interface to a persistent Mongo file.
+
+    The MongoDict inherits from :class:`~core.synced_collection.SyncedCollection`
+    and :class:`~core.syncedlist.SyncedList`.
+
+    .. code-block:: python
+
+        synced_list = MongoList('data')
+        synced_list.append("bar")
+        assert synced_list[0] == "bar"
+        assert len(synced_list) == 1
+        del synced_list[0]
+
+    .. warning::
+
+        While the MongoList object behaves like a list, there are
+        important distinctions to remember. In particular, because operations
+        are reflected as changes to an underlying database, copying (even deep
+        copying) a MongoList instance may exhibit unexpected behavior. If a
+        true copy is required, you should use the `to_base()` method to get a
+        dictionary representation, and if necessary construct a new MongoList
+        instance: `new_list = MongoList(old_list.to_base())`.
+
+    Parameters
+    ----------
+    name: str
+        The name of the  collection (Default value = None).
+    client:
+        A Mongo client.
+    Mongo_kwargs: dict
+        kwargs arguments passed through to the `pymongo.MongoClient` function.
+    database : string
+        Name of database (Default value = 'signac_db').
+    collection : string
+        Name of collection (Default value = 'collection')
+    data: mapping, optional
+        The intial data pass to MongoList. Defaults to `list()`
+    parent: object, optional
+        A parent instance of MongoList or None (Default value = None).
+    """
+
+    pass
```

Comment:
Not sure if we even need the `pass` statement, but there is inconsistent spacing here compared to the previous class.